### PR TITLE
fix bug in metadata.py

### PR DESCRIPTION
**Pull request by jlippuner**
This small bug left temporary files in my local backend

### CODE REVIEW

File: src/s3ql/metadata.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def cycle_metadata(backend, keep=10):
         pass
     cycle_fn("s3ql_metadata_new", "s3ql_metadata")
 
-    if cycle_fn is backend.copy:
+    if not backend.has_native_rename:
         backend.delete('s3ql_metadata_new')
```
Comment (Nikratio) [inline src/s3ql/metadata.py]: If `cycle_fn` is not `backend.copy`, then it is `backend.rename`, so there should be no `s3ql_metadata_new` object (it has been renamed). Thus the current code seems ok.

Looking at your proposed change, note that if `not backend.has_native_rename`, then `cycle_fn is backend.copy` (per line 115-118), so your change doesn't actually change anything in practice.

Comment (jlippuner) [inline src/s3ql/metadata.py]: If I remember correctly, i found that the test `cycle_fn is backend.copy` returned False even when it was actually set to `backend.copy` in lines 115-118, which is why i changed it to checking `backend.has_native_rename`, which produced the correct result.

This change did make a difference in my setup. i used a local S3QL in an acd_cli mount point. Before this change, i would have a `s3ql_metadata_new` file left over after running `mkfs.s3ql` and after the change this temporary file would be correctly deleted.

**Comment by Nikratio** [comment]
Ah, good catch. I didn't realize this was a problem. But I think there's a better way to fix the problem. Could you try if the change that I just made still works for you?

**Comment by Nikratio** [comment]
Merged my version. Let me know in case it doesn't work for you. Thanks!

**Pull request closed without merging**